content-learn/mastering-meshery-istio/performance-of-service-mesh/istio/fault-injection.mdx

```diff
@@ -0,0 +1,188 @@
+
+```sh
+kubectl get virtualservice ratings -o yaml
+```
+
+Config:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+    - ratings
+  http:
+    - fault:
+        delay:
+          fixedDelay: 7s
+          percentage:
+            value: 100
+      match:
+        - headers:
+            end-user:
+              exact: jason
+      route:
+        - destination:
+            host: ratings
+            subset: v1
+    - route:
+        - destination:
+            host: ratings
+            subset: v1
+```
+
+Now we login to `/productpage` as user `jason` and observe that the page loads but because of the induced delay between services the reviews section will show :
+
+<b>Error fetching product reviews!</b>
+Sorry, product reviews are currently unavailable for this book.
+
+If you logout or login as a different user, the page should load normally without any errors.
+
+<h2 class="chapter-sub-heading">
+  {" "}
+  Inject a route rule to create a fault using HTTP abort
+</h2>
+
+In this section, , we will introduce an HTTP abort to the ratings microservices for user `jason`.
+
+Using Meshery, navigate to the Istio management page:
+
+1. Enter `default` in the `Namespace` field.
+2. Click the (+) icon on the `Apply Custom Configuration` card and paste the configuration below.
+
+This will update the existing virtual service definition for ratings to inject a HTTP abort for user `jason` to access the ratings V1.
+
+In a few, we should be able to verify the virtual service by using the command below:
+
+```sh
+kubectl get virtualservice ratings -o yaml
+```
+
+Config:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+    - ratings
+  http:
+    - fault:
+        abort:
+          httpStatus: 500
+          percentage:
+            value: 100
+      match:
+        - headers:
+            end-user:
+              exact: jason
+      route:
+        - destination:
+            host: ratings
+            subset: v1
+    - route:
+        - destination:
+            host: ratings
+            subset: v1
+```
+
+Now we login to `/productpage` as user `jason` and observe that the page loads without any new delays but because of the induced fault between services the reviews section will show:
+
+`Ratings service is currently unavailable`.
+
+<h3 class="chapter-sub-heading"> Verify fault injection</h3>
+
+Verify the fault injection by logging out (or logging in as a different user), the page should load normally without any errors.
+
+<br />
+<h3>Alternative: Manual installation</h3>
+Follow these steps if the above steps did not work
+<br />
+<br />
+
+<h4 class="chapter-alt-heading">
+  Route all traffic to version V1 of all services
+</h4>
+
+```sh
+kubectl apply -f samples/bookinfo/networking/virtual-service-all-v1.yaml
+```
+
+<h4 class="chapter-alt-heading">
+  {" "}
```

Comment:
Any specific reason for this `{" "}` ?

Reply:
Prettier did what its supposed to do 😄

Reply:
Then, please overwrite those. That's not needed at all. We prefer not to use prettier, as we are already using eslint which does take care of the formatting's.